### DISCUSSION TITLE
Post slugs not logged from `getStaticprops` on Next.js ISR

### DISCUSSION THREAD

**Discussion by zakiego** [Thinkmill/keystatic#197] [Q&A] [answered]
Reader works during the first build process, but when doing ISR, the reader becomes empty. I'm just curious, what causes this to happen?

Here is the code I use on getStaticProps:

```ts
const getAllPost = async () => {
  const postSlugs = await reader.collections.posts.list();

  console.log("postSlugs", postSlugs);

  const data = await Promise.all(
    postSlugs.map(async (slug) => {
      const post = await reader.collections.posts.read(slug);
      const content = (await post?.content()) || [];
      return {
        ...post,
        content: content,
        slug,
      };
    }),
  );

  return data;
};
```

During the build process in Vercel:

<img width="482" alt="image" src="https://user-images.githubusercontent.com/78015359/228318113-08263aca-aecc-4b09-a1b2-32e62c581b16.png">

Log function when ISR in Vercel:

<img width="333" alt="image" src="https://user-images.githubusercontent.com/78015359/228318216-ec231abc-25aa-4733-b442-050949bcc088.png">

Thank you ♥️

**Comment by ccssmnn** [comment, marked as answer]
After deploying your app on Vercel, the functions won't have access to the file system where Keystatic stores your content. When all your content is managed by Keystatic you probably don't need ISR. Rather redeploy your site on content changes, which are basically new commits to your repo since they are file changes.

When you want to add dynamic data to your pages, consider middleware or client side data fetching.

I hope this helps :smile:
